File: 26/45.cpp
#include <fstream>
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	ifstream F;
	string path = "C:\\Users\\damikh13\\Desktop\\26data\\26-45.txt";
	F.open(path);

	int n; F >> n;
	vector<int> even;
	vector<int> odd;
    vector<int> all_;
	for (int i = 0; i < n; ++i) {
		int x; F >> x;
		if (x % 2 == 0) {
			even.push_back(x);
		}
		else {
			odd.push_back(x);
		}
        all_.push_back(x);
	}

	sort(all_.begin(), all_.end());

	int cnt = 0, max_avg = -1;
	for (int i = 0; i < even.size() - 1; ++i) {
		for (int j = i + 1; j < even.size(); ++j) {
			int avg = (even[i] + even[j]) / 2;
			int l = 0, r = all_.size() - 1;
			while (l < r) {
				int middle = (l + r) / 2;
				if (avg > all_[middle]) {
					l = middle + 1;
				}
				else {
					r = middle;
				}
			}
			if (all_[r] == avg) {
				max_avg = max(max_avg, avg);
				++cnt;
			}
		}
	}
	
	for (int i = 0; i < odd.size() - 1; ++i) {
		for (int j = i + 1; j < odd.size(); ++j) {
			int avg = (odd[i] + odd[j]) / 2;
			int l = 0, r = all_.size() - 1;
			while (l < r) {
				int middle = (l + r) / 2;
				if (avg > all_[middle]) {
					l = middle + 1;
				}
				else {
					r = middle;
				}
			}
			if (all_[r] == avg) {
				max_avg = max(max_avg, avg);
				++cnt;
			}
		}
	}

    cout << cnt << ' ' << max_avg << '\n';
}

File: 26/45.py
File = open(r"C:\Users\damikh13\Desktop\26data\26-45.txt", "r")
S = File.readlines()
n = int(S[0])

even = list()
odd = list()

for i in range(1, len(S)):
    x = int(S[i])
    if x % 2 == 0:
        even.append(x)
    else:
        odd.append(x)

even = sorted(even)
odd = sorted(odd)

def contains(lst, x):
    if x > lst[-1]:
        return False
    l = 0
    r = len(lst)
    while l < r:
        mid = (l + r) // 2
        if x > lst[mid]:
            l = mid + 1
        else:
            r = mid
    if lst[r] == x:
        return True
    else:
        return False

avg_max = -1
cnt = 0
for i in range(len(even) - 1):
    for j in range(i + 1, len(even)):
        x = even[i]
        y = even[j]
        avg = (x + y) // 2
        if contains(even, avg) or contains(odd, avg):
            avg_max = max(avg_max, avg)
            cnt += 1

for i in range(len(odd) - 1):
    for j in range(i + 1, len(odd)):
        x = odd[i]
        y = odd[j]
        avg = (x + y) // 2
        if contains(even, avg) or contains(odd, avg):
            avg_max = max(avg_max, avg)
            cnt += 1

print(cnt, avg_max)